test/test_coretasks.py

```diff
@@ -4,128 +4,112 @@
-    assert sopel.channels["#test"].privileges[Identifier("Uop")] == OP
+    Sopel 6.6.6 and older did not handle this correctly.
 
+    .. seealso::
 
-def test_bot_mixed_mode_types(sopel):
-    """
-    Ensure mixed argument- and non-argument- modes are handled
-    Sopel 6.6.6 and older did not behave well. #1575
+        Github Issue #1575.
     """
+    irc = IRCFactory(mockbot)
+    irc.channel_joined('#test', ['Uvoice', 'Uop'])
 
-    # RPL_NAMREPLY to create Users and (zeroed) privs
-    for user in set("Uvoice Uop Uadmin Uvoice2 Uop2 Uadmin2".split(" ")):
-        pretrigger = PreTrigger(
-            "Foo", ":test.example.com 353 Foo = #test :Foo %s" % user
-        )
-        trigger = Trigger(sopel.config, pretrigger, None)
-        coretasks.handle_names(MockSopelWrapper(sopel, trigger), trigger)
-
-    # Non-attribute-requiring non-permission mode
-    pretrigger = PreTrigger("Foo", "MODE #test +amov Uadmin Uop Uvoice")
-    trigger = Trigger(sopel.config, pretrigger, None)
-    coretasks.track_modes(MockSopelWrapper(sopel, trigger), trigger)
-
-    assert sopel.channels["#test"].privileges[Identifier("Uvoice")] == VOICE
-    assert sopel.channels["#test"].privileges[Identifier("Uop")] == OP
-    assert sopel.channels["#test"].privileges[Identifier("Uadmin")] == ADMIN
-
-    # Attribute-requiring non-permission modes
-    # This results in a _send_who, which isn't supported in MockSopel or this
-    # test, so we just make sure it results in an exception instead of privesc.
-    pretrigger = PreTrigger("Foo", "MODE #test +abov Uadmin2 x!y@z Uop2 Uvoice2")
-    trigger = Trigger(sopel.config, pretrigger, None)
-    try:
-        coretasks.track_modes(MockSopelWrapper(sopel, trigger), trigger)
-    except AttributeError as e:
-        if e.args[0] == "'MockSopel' object has no attribute 'enabled_capabilities'":
-            return
-
-    assert sopel.channels["#test"].privileges[Identifier("Uvoice2")] == VOICE
-    assert sopel.channels["#test"].privileges[Identifier("Uop2")] == OP
-    assert sopel.channels["#test"].privileges[Identifier("Uadmin2")] == ADMIN
-
-
-def test_mode_colon(sopel):
-    """
-    Ensure mode messages with colons are parsed properly
+    irc.mode_set('#test', '+qao', ['Uvoice', 'Uvoice', 'Uvoice'])
+    assert mockbot.channels["#test"].privileges[Identifier("Uop")] == 0
+    assert mockbot.channels["#test"].privileges[Identifier("Uvoice")] == (
+        ADMIN + OWNER + OP), 'Uvoice got +q, +a, and +o modes'
+
+    irc.mode_set('#test', '-o+o-qa+v', [
+        'Uvoice', 'Uop', 'Uvoice', 'Uvoice', 'Uvoice'])
+    assert mockbot.channels["#test"].privileges[Identifier("Uop")] == OP, (
+        'OP got +o only')
+    assert mockbot.channels["#test"].privileges[Identifier("Uvoice")] == VOICE, (
+        'Uvoice got -o, -q, -a, then +v')
+
+
+def test_bot_mixed_mode_types(mockbot):
+    """Ensure mixed argument- and non-argument- modes are handled.
```

Comment:
```suggestion
    """Ensure mixed argument-required and -not-required modes are handled.
```

It's still weird, but less so. 🙁

Reply:
It was there before me, and I'm still not sure I understand what it says. So...